src/connector/src/parser/mod.rs

```diff
@@ -916,12 +920,14 @@ impl SpecificParserConfig {
                     config.topic = get_kafka_topic(props)?.clone();
                     config.client_config = SchemaRegistryAuth::from(props);
                 } else {
-                    config.aws_auth_props = Some(
-                        serde_json::from_value::<AwsAuthProps>(
-                            serde_json::to_value(props).unwrap(),
-                        )
-                        .map_err(|e| anyhow::anyhow!(e))?,
-                    );
+                    config.aws_auth_props = row_options
+                        .map(|options| {
+                            serde_json::from_value::<AwsAuthProps>(
+                                serde_json::to_value(options).unwrap(),
+                            )
+                            .map_err(|e| anyhow::anyhow!(e))
+                        })
+                        .transpose()?;
```

Comment:
This looks too ad-hoc a solution to only solve the aws auth config issue. 

Specifically,
- How can other people know which one to use?
- Since `props` is still merged with `row_options`, does it have other possible misuses? It still looks very unsafe.

Reply:
> This looks too ad-hoc a solution to only solve the aws auth config issue.

This pr is actually an ad-hoc fix for the aws auth config issue. Completely resolving the misuse between props and options requires a huge refactor, maybe @tabVersion can share some thoughts on it.

> How can other people know which one to use?

AWS props within `WITH` are used to access kinesis, while those within `FORMAT ENCODE` are used to access schema definition in s3. Maybe we can note it in the doc.

> Since props is still merged with row_options, does it have other possible misuses? It still looks very unsafe.

So far we can only come up with this case of misuse :)

Reply:
> > This looks too ad-hoc a solution to only solve the aws auth config issue.
> 
> This pr is an ad-hoc fix for the aws auth config issue indeed. Completely resolving the misuse between props > and options requires a huge refactor, maybe @tabVersion can share some thoughts on it.

The solution is acceptable to me because only encode options need auth params in this part. Connector props' auth part will be on the compute node when building actor. @Rossil2012 can you write some comments about this? 

> > How can other people know which one to use?
> 
> AWS props within WITH are used to access kinesis, while those within FORMAT ENCODE are used to access schema definition in s3. Maybe we can note it in the doc.

The only part of connector props used when infering schema is the `connector` name I think.

Reply:
> AWS props within `WITH` are used to access kinesis, while those within `FORMAT ENCODE` are used to access schema definition in s3. Maybe we can note it in the doc.

This confusion was the reason #12858 (protobuf sink) did not support reading definition from s3. I was under the conception that aws credentials are expected to only be provided once and it needs to be part of `WITH` because of kinesis. If we do decide to keep two separate copies that is allowed to be different, it should definitely be noted down.